tests/phpunit/tests/html-api/generate-html5-named-character-references.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+require_once __DIR__ . '/../../../../src/wp-includes/class-wp-token-map.php';
+
+/**
+ * Stores a mapping from HTML5 named character reference to its transformation metadata.
+ *
+ * Example:
+ *
+ *     $entities['&copy;'] === array(
+ *         'codepoints' => array( 0xA9 ),
+ *         'characters' => '©',
+ *     );
+ *
+ * @see https://html.spec.whatwg.org/entities.json
+ *
+ * @var array.
+ */
+$entities = json_decode(
+	file_get_contents( __DIR__ . '/../../data/html5-entities.json' ),
+	JSON_OBJECT_AS_ARRAY
+);
+
+/**
+ * Direct mapping from character reference name to UTF-8 string.
+ *
+ * Example:
+ *
+ *     $character_references['&copy;'] === '©';
+ *
+ * @var array.
+ */
+$character_references = array();
+foreach ( $entities as $reference => $metadata ) {
+	$reference_without_ampersand_prefix                          = substr( $reference, 1 );
+	$character_references[ $reference_without_ampersand_prefix ] = $metadata['characters'];
+}
+
+$html5_map       = WP_Token_Map::from_array( $character_references );
+$module_contents = <<<EOF
+<?php
+
+/**
+ * Auto-generated class for looking up HTML named character references.
+ *
+ * To regenerate, run the generation script directly.
+ *
+ * Example:
+ *
+ *     php tests/phpunit/tests/html-api/generate-html5-named-character-references.php
+ *
+ * @package WordPress
+ * @since 6.6.0
+ */
+
+// phpcs:disable
+
+global \$html5_named_character_references;
```

Comment:
Note for other reviewers:

It took me a while to see that `\$var` is escaping the `$` in `$var`. I kept seeing this as the namespace global space. In the generated file is just `global $html5_named_character_references;`

Reply:
Added a note in a comment above the template. Hope that helps